### PR TITLE
(session) compose session hash

### PR DESCRIPTION
**Pull request by jackturnbull**
Currently seeing a compiler bug when attempting to compile with raven.cr. Although I'm not convinced entirely that this is an issue with action-controller, it does seem to be considered good practice to stick with composition rather than inheritence for this sort of stuff (https://github.com/crystal-lang/crystal/issues/3238#issuecomment-244548787).

The approach taken was effectively the same as amberframework/amber#930 which fixed the same issue within Amber.

Previously, the following:
```crystal
require "raven"
require "action-controller"

Raven.configure do |settings|
  settings.async = true
end
```
Causes the compiler panic:
```
Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues
Module validation failed: Function return type does not match operand type of return inst!
  ret %"(Array(AnyHash::JSONTypes::Value) | Bool | Float32 | Float64 | Hash(String | Symbol, AnyHash::JSONTypes::Value)+ | Int128 | Int16 | Int32 | Int64 | Int8 | JSON::Any | Set(AnyHash::JSONTypes::Value) | String | Symbol | Time | UInt128 | UInt16 | UInt32 | UInt64 | UInt8 | Nil)" %1, !dbg !21
 %"(Array(AnyHash::JSONTypes::Value) | Bool | Float32 | Float64 | Hash(String | Symbol, AnyHash::JSONTypes::Value) | Int128 | Int16 | Int32 | Int64 | Int8 | JSON::Any | Set(AnyHash::JSONTypes::Value) | String | Symbol | Time | UInt128 | UInt16 | UInt32 | UInt64 | UInt8 | Nil)" = type { i32, [3 x i64] } (Exception)
  from Crystal::CodeGenVisitor#finish:Nil
  from Crystal::Compiler#codegen<Crystal::Program, Crystal::ASTNode+, Array(Crystal::Compiler::Source), String>:(Tuple(Array(Crystal::Compiler::CompilationUnit), Array(String)) | Nil)
  from Crystal::Compiler#compile<Array(Crystal::Compiler::Source), String>:Crystal::Compiler::Result
  from Crystal::Command#run_command<Bool>:Nil
  from Crystal::Command#run:(Bool | Crystal::Compiler::Result | Nil)
  from main
```

### CODE REVIEW

File: src/action-controller/session.cr [hunk 1]
```diff
@@ -22,6 +22,9 @@ class ActionController::Session < Hash(String, String | Int64 | Float64 | Bool)
   getter modified : Bool
   property domain : String?
   @encoder : MessageEncryptor | MessageVerifier
+  @store : Hash(String, String | Int64 | Float64 | Bool)
+
+  forward_missing_to @store
 
   def initialize
     super
```
Comment (stakach) [inline src/action-controller/session.cr]: I think we can remove this call to `super`

**Comment by stakach** [comment]
Looks like a great solution - had no idea `forward_missing_to` was a feature

**Comment by stakach** [comment]
Can you remove the call to super and I'll merge

File: src/action-controller/session.cr (hunk 1 above)
Comment (jackturnbull) [reply src/action-controller/session.cr]: Good catch!

**Comment by jackturnbull** [comment]
Thanks for getting onto it so quickly! Didn't expect that at all :)

**Pull request merged**